components/FlexibleHeader/src/MDCFlexibleHeaderView.m

```diff
@@ -776,17 +785,24 @@ - (void)fhv_accumulatorDidChange {
   if (!_trackingScrollView) {
     // Set the shadow opacity directly.
     self.layer.shadowOpacity = _visibleShadowOpacity;
-
     return;
   }
 
   CGRect frame = self.frame;
 
   CGFloat frameBottomEdge = [self fhv_projectedHeaderBottomEdge];
   frameBottomEdge = MAX(0, MIN(kShadowScaleLength, frameBottomEdge));
-
   CGFloat boundedAccumulator = MIN([self fhv_accumulatorMax], _shiftAccumulator);
 
+  if (_shiftBehavior == MDCFlexibleHeaderShiftBehaviorEnabled) {
+    CGFloat contentHeight = self.computedMinimumHeight - MDCDeviceTopSafeAreaInset();
+    CGFloat hideThreshold = kContentHidingThreshold;
+    CGFloat alpha = MAX(contentHeight - boundedAccumulator / hideThreshold, 0) / contentHeight;
+    for (UIView *view in _viewsToHideWhenShifted) {
+      view.alpha = alpha;
+    }
+  }
```

Comment:
This logic may be simplified if you use the phase properties and move this logic to after `fhv_recalculatePhase` is invoked in `fhv_commitAccumulatorToFrame`. You should then be able to use the `_scrollPhasePercentage` to set the view alpha.

Reply:
I don't think I can use `_scrollPhasePercentage` for this, take a look:

![ezgif-4-c113d5b396](https://user-images.githubusercontent.com/4545451/32282381-cb2fe1f6-bef7-11e7-8d5a-f9d9792ad80b.gif)

Reply:
Hrm, weird. The documented behavior is for shifting is supposed to be: "0 is unshifted, 1.0 is fully shifted off-screen". Seems there might be a bug in the shift phase calculations?

Reply:
The configurator demo looks like it's setting the values as expected....I'm not sure why it would be working differently here.

Reply:
Weird indeed, I'm seeing similar numbers in the Configurator demo

Reply:
Must be something recent? I can't repro the odd numbers on the latest internal cut of MDC.

Reply:
I just rebased and this is still happening, `_scrollPhasePercentage` doesn't go from 0 to 1. Can I go ahead with the current implementation?

Reply:
👍